src/hera/workflows/validators.py

```diff
@@ -43,30 +43,47 @@ def validate_name(name: str, max_length: Optional[int] = None, generate_name: bo
     return name
 
 
-def validate_storage_units(value: str) -> None:
-    """Validates the units of the given value.
+def validate_binary_units(value: str) -> None:
```

Comment:
renamed from `validate_storage_units` to `validate_binary_units` as this applies to memory and ephemeral volumes

Reply:
Unfortunately this helper is public and we don't know who might use it outside of the core of Hera. Any chance we can leave `validate_storage_units(value: str) -> None` as public and internally it passes the value to `validate_binary_units`? That way we get the new function with improved functionality while maintaining backwards compatibility

Reply:
Thanks for your quick review!

In the commit below I kept the public function and use aliases to refer to the `_validate_binary_units`, which is now made private so that we can the implementaion later if necessary.

To avoid confusion and also to make the function names consistent, other function names are now named more explicitly for their use cases (`validate_storage_units`, `validate_cpu_units`, `validate_memory_units`) as well as converters.

https://github.com/argoproj-labs/hera/pull/944/commits/31184f84447e6129ab99bdf008306d84ab34b9d7